core/base/inc/Rtypes.h

```diff
@@ -225,44 +225,120 @@ namespace ROOT {
 
 } // namespace ROOT
 
-// The macros below use TGenericClassInfo, so let's ensure it is included
+// The macros below use TGenericClassInfo and TInstrumentedIsAProxy, so let's
+// ensure they are included.
 #ifndef ROOT_TGenericClassInfo
 #include "TGenericClassInfo.h"
 #endif
+#ifndef ROOT_TIsAProxy
+#include "TIsAProxy.h"
+#endif
+
+
+// Parts or ROOT are compiled with no-rtti and cannot use or even see typeid().
+// Hide the following declaration (that is only needed by ClassDefInline) for
+// these cases by #defining R__NO_INLINE_CLASSDEF:
+#ifndef R__NO_INLINE_CLASSDEF
+namespace ROOT {
+   template <typename T>
+   class ClassDefGenerateInitInstanceLocalInjector {
+      static TClass* fgIsA;
+   public:
+      static void *New(void *p) { return p ? new(p) T : new T; };
+      static void *NewArray(Long_t nElements, void *p) {
+         return p ? new(p) T[nElements] : new T[nElements]; }
+      static void Delete(void *p) { delete ((T*)p); }
+      static void DeleteArray(void *p) { delete[] ((T*)p); }
+      static void Destruct(void *p) { ((T*)p)->~T();  }
+      static ::ROOT::TGenericClassInfo *GenerateInitInstanceLocal() {
+         T *ptr = 0;
+         static ::TVirtualIsAProxy* isa_proxy = new ::TInstrumentedIsAProxy<T>(0);
+         static ::ROOT::TGenericClassInfo
+            R__instance(T::Class_Name(), T::Class_Version(),
+                        T::DeclFileName(), T::DeclFileLine(),
+                        typeid(T), ROOT::Internal::DefineBehavior(ptr, ptr),
+                        &T::Dictionary, isa_proxy, 0, sizeof(T) );
+         R__instance.SetNew(&New);
+         R__instance.SetNewArray(&NewArray);
+         R__instance.SetDelete(&Delete);
+         R__instance.SetDeleteArray(&DeleteArray);
+         R__instance.SetDestructor(&Destruct);
+         return &R__instance;
+      }
+      static void Dictionary() { fgIsA = GenerateInitInstanceLocal()->GetClass(); }
```

Comment:
It was the right commit and I am seeing:

```
   // Trigger autoloading if dictionary is split
   if (autoLoad)
      dictStream << "   gInterpreter->EnableAutoLoading();\n"
                 << "   gInterpreter->AutoLoad(\"" << fullname << "\");\n";
```

and did not pay enough attention to the comment and test.

And indeed you are right, this is not needed here since this code is to insure that part 1 of a 2 parts dictionary is loaded with part 2.

Thanks,
Philippe.